semantics/language/common/compat.k

```diff
@@ -240,4 +242,6 @@ module COMPAT
      rule dotK => .K [structural]
      rule lookup(Key |-> Val:K _:Map, Key:K) => Val
 
+     rule isKItem(K:KItem) => true
```

Comment:
Can you try without these two rules. I think it might work based on default sort handling.